backfill.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+# this script is used to update the article-json stored in lax
+# performing just an INGEST of content, *no* PUBLISH events are
+# sent whatsoever.
+
+# talking to lax via the adaptor.py script for many thousands of articles is
+# extremely slow, so this script bypasses all that, bulk generates articles,
+# bulk validates them then tells lax to do a bulk ingest.
+
+set -euo pipefail # strict mode
+#set -xv # debugging
+
+# housekeeping
+
+trap ctrl_c INT
+function ctrl_c() {
+    echo "caught ctrl-c"
+    exit 1
+}
+
+srcdir=$(pwd)
+errcho(){ >&2 echo $@; }
+
+# where articles will be linked to/downloaded for backfill
+defaultrunpath="run-$(date +'%Y%m%d%H%M%S')"
+runpath=${1:-$defaultrunpath}
+runpath=$(realpath $runpath) # realpath needs to be installed :(
+mkdir -p "$runpath"
+
+shift # download-api-raml.sh takes an arg :P
```

Comment:
big badda boom: https://github.com/elifesciences/bot-lax-adaptor/blob/develop/download-api-raml.sh#L8

took me a while to find it as well. I didn't think subsequently sourced scripts would have access to params passed to parent, but it turns out they do. what `shift` does is pop the first arg off of `$@` and return it. in this case I don't capture the return value and downstream scripts don't see the first arg.